pkg/apis/core/types_shoot.go

```diff
@@ -497,6 +497,37 @@ type KubeAPIServerConfig struct {
 	EnableAnonymousAuthentication *bool
 	// EventTTL controls the amount of time to retain events.
 	EventTTL *metav1.Duration
+	// AccessControl provides authorization mechanisms for Shoot-API-Server.
+	AccessControl *AccessControl
+}
+
+// AccessControl provides authorization mechanisms for Shoot-API-Server.
+// Note: The schema (incl. child structs) and documentation resembles istio's AuthorizationPolicy.
+type AccessControl struct {
+	// The action to take on the source of request.
+	Action *AuthorizationAction
```

Comment:
Thanks for highlighting. I have added comments on the optionality of the fields. Following up on default values, and speaking greenfield: I would have thought to keep this entirely optional without default values for access control. I'd argue there is nothing to allow or whitelist at all if nothing was provided in the shoot spec. 

Happy to change obviously, but just would like to understand the rational (?)

Reply:
`AccessControl` is optional (and it should be). What I do not understand is why you made `Action` and `Source` optional. According to your validation logic in `pkg/apis/core/validation/shoot.go` neither can actually be `nil` if `AccessControl` is specified.
Why do you not just get rid of the pointer? Do you foresee other options that could be used instead of `Action`/`Source`?

Reply:
Thanks, I see. I had it optional as I thought to see a convention to make all fields optional in `pkg/apis/core/types_shoot.go`. I have changed according to your suggestion and made `Action` and `Source` optional fields. Validation should incorporate this as well.